ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/home/viewmodel/WalletViewModel.kt

```diff
@@ -172,6 +172,20 @@ class WalletViewModel(application: Application) : AndroidViewModel(application)
             null
         )
 
+    @OptIn(ExperimentalCoroutinesApi::class)
+    val transactionHistory = synchronizer
+        .flatMapLatest {
+            if (null == it) {
+                flowOf(emptyList())
+            } else {
+                it.toTransactionHistory()
+            }
+        }.stateIn(
+            viewModelScope,
+            SharingStarted.WhileSubscribed(ANDROID_STATE_FLOW_TIMEOUT),
+            emptyList()
```

Comment:
Let's emit a sealed class with a Loading state.  That would allow the UI to show a progress indicator instead of thinking the list is empty.

Reply:
Sealed class added. The Loading state in the UI is implemented as well. Thank you.